executor/builder.go

```diff
@@ -1599,10 +1599,25 @@ func (b *executorBuilder) buildSort(v *plannercore.PhysicalSort) Executor {
 	if b.err != nil {
 		return nil
 	}
+	columnIdxsUsedByChild, columnMissing := retrieveColumnIdxsUsedByChild(v.Schema(), v.Children()[0].Schema())
+	if columnIdxsUsedByChild != nil && columnMissing {
+		// In the expected cases colMissing will never happen.
+		// However, suppose that childSchema contains generatedCol and is cloned by selfSchema.
```

Comment:
What does "generatedCol" mean ?

I found that `columnMissing` happens on `show` executor, such as `show status like 'Handler_read_rnd_next'`, but I just saw ordinary columns in schema of `show`.

_Besides, I found that failure of `show status like 'Handler_read_rnd_next'` is caused by a projection with new `UniqueID` columns (see [here](https://github.com/pingcap/tidb/blob/2f33d57db1e72d03929b76b9481ddf31aed3d762/planner/core/planbuilder.go#L2029)), which would be eliminated by "projection elimination", and lead to inconsistence of schema between `sort` and `show (+selection)`. I think modify schema of this projection would avoid the "columnMissing"._

Reply:
`generatedCol` means it's not from a real table, `UniqueID` of those columns would be `0`. 
Yes, I notice that too. You means use old `UniqueID` to new columns(cloned) is okay?

Reply:
I did a quick try that comment out `newCol.UniqueID = b.ctx.GetSessionVars().AllocPlanColumnID()`, and it worked.
I am not quite sure that `UniqueID` of projection can be the same as `Show`.

What do you think about this ? @SunRunAway

Reply:
@pingyu @SunRunAway If we can't find out it, we may leave more comments to help other developer in future.